components/eventing-controller/pkg/handlers/beb.go

```diff
@@ -98,7 +98,7 @@ func (b *Beb) SyncSubscription(subscription *eventingv1alpha1.Subscription, clea
 
 	// get the internal view for the ev2 subscription
 	var statusChanged = false
-	sEv2, err := getInternalView4Ev2(subscription, apiRule, b.WebhookAuth, b.ProtocolSettings, b.Namespace, b.SubNameMapper)
+	sEv2, err := getInternalView4Ev2(subscription, apiRule, *b.WebhookAuth, b.ProtocolSettings, b.Namespace, b.SubNameMapper)
```

Comment:
Why could we not continue to use a "pointer" in the signature and instead we are copying (not deep copy) the object where the pointer points. The usual convention is to use in this case pointers (runtime performance and also avoiding of errors related to the "copy" of objects implementation in go) and only dereferencing the pointer at the moment of usage. At this point perhaps a "copy" of the object could be useful, like in our case.

Reply:
Sure, I'll make it a separate PR.